k4Gen/CMakeLists.txt

```diff
@@ -6,12 +6,11 @@ find_package(HepMC)
 find_package(Pythia8 COMPONENTS pythia8 pythia8tohepmc)
 find_package(HepPDT)
 find_package(EvtGen)
-find_package(Gaudi)
 
 file(GLOB k4gen_plugin_sources src/components/*.cpp)
 gaudi_add_module(k4Gen
                  SOURCES ${k4gen_plugin_sources}
-                 LINK Gaudi::GaudiKernel HepMC Gaudi::GaudiAlgLib k4FWCore::k4FWCore k4FWCore::k4FWCorePlugins HepPDT ${EVTGEN_LIBRARIES} EDM4HEP::edm4hep EDM4HEP::edm4hepDict)
+                 LINK Gaudi::GaudiKernel ${HEPMC_LIBRARIES} Gaudi::GaudiAlgLib k4FWCore::k4FWCore k4FWCore::k4FWCorePlugins ${HEPPDT_LIBRARIES} ${EVTGEN_LIBRARIES} EDM4HEP::edm4hep EDM4HEP::edm4hepDict)
```

Comment:
What setup are you using here? This works perfectly fine for me with the `2021-03-26` release:
-- Found HepMC: /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/hepmc-2.06.10-mo3ujzg5t44bok3aioudbzv3trajus6t/include  
-- Found Pythia8: /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/pythia8-8303-uwum425ziw4i3gxrhludqhcwlbw65qun/include  
-- Found HepPDT: /cvmfs/sw.hsf.org/spackages/linux-centos7-broadwell/gcc-8.3.0/heppdt-2.06.01-xv236he5m4zmmhrfdytvhk44rxovbghg/include

Reply:
I built this locally, and the `Find*.cmake` that are shipped with `k4Gen` are used to detect `HepMC` and `HepPDT` in this case. These do not define targets. Are there other `Find*.cmake` used in the stack?

Reply:
Ah, I see. The [HSF cmaketools](https://github.com/hsf/cmaketools) are also installed, but not widely used, so at first glance I thought there might be a difference there. The current `LINK` line just adds  verbatim `-lHepMC -lHepPDT` link flags, which happen to work with the cvmfs setup, but indeed it is better to use the defined variables.